pbjam/plotting.py

```diff
@@ -284,18 +317,108 @@ def plot_spectrum(self, pg=None, path=None, ID=None, savefig=False):
         ax.set_xlabel(r'Frequency ($\mu \rm Hz$)')
         ax.set_ylabel(r'SNR')
         ax.legend(loc=1)
-
+        
+        fig.tight_layout()
         if savefig:
-            outpath = os.path.join(*[path, f'{type(self).__name__}_{str(ID)}.png'])
-            fig.savefig(outpath)
+            self._save_my_fig(fig, 'spectrum', path, ID)
 
         return fig
 
-    # Asy_peakbag
+
+    def _fill_diag(self, axes, vals, vals_err, idxs):
```

Comment:
Missing documentation.  Are there units tests for this (if yes, can probably cope without docs as I think I understand the function name. )

Reply:
Updated the PR with some template docstring blocks. These will be filled out soon. 

No unit tests yet, feel free to add some :P